apps/desktop/src/app/sync/sources/s3.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  ConnectorSettings,
+  ISourceConnector,
+  SourceConnectorDefinition,
+  SyncItem,
+  SearchResults,
+  ConnectorParameters,
+  FileStatus,
+  Field,
+} from '../models';
+import { _Object, S3Client, ListObjectsV2Command, GetObjectCommand } from '@aws-sdk/client-s3';
+import { Observable, of, from, map, switchMap } from 'rxjs';
+
+const MAX_PAGE_SIZE = 1000;
+
+export const S3Connector: SourceConnectorDefinition = {
+  id: 's3',
+  title: 'AWS S3',
+  logo: 'assets/logos/s3.svg',
+  description: 'Object storage service developed by Amazon',
+  factory: (data?: ConnectorSettings) => of(new S3Impl(data)),
+};
+
+class S3Impl implements ISourceConnector {
+  hasServerSideAuth = false;
+  resumable = false;
+  client?: S3Client;
+  bucket?: string;
+
+  constructor(data?: ConnectorSettings) {
+    // eslint-disable-next-line no-empty-function
+  }
+
+  getParameters(): Observable<Field[]> {
+    return of([
+      {
+        id: 'region',
+        label: 'Region',
+        type: 'text',
+        help: 'e.g. us-east-1',
+        required: true,
+      },
+      {
+        id: 'bucket',
+        label: 'Bucket',
+        type: 'text',
+        required: true,
+      },
+      {
+        id: 'access_key_id',
+        label: 'Access key id',
+        type: 'text',
+        required: true,
+      },
+      {
+        id: 'secret_access_key',
+        label: 'Secret access key',
+        type: 'text',
+        required: true,
+      },
+    ]);
```

Comment:
should we translate the labels from this list?

Reply:
I added translations for `Region`.
Not sure if `Access key id` and 'Secret access key' should be translated or not.
In the case of `Bucket` it's the same for all langs.